File: Numbers-Teach-Android/app/src/main/java/com/dferreira/numbers_teach/preferences/PreferencesUtils.java
package com.dferreira.numbers_teach.preferences;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.text.TextUtils;

/**
 * Utils to provide the values of the preferences selected
 */
public class PreferencesUtils {
    /**
     * @param context Get the delay betweenSlides
     * @return Delay in milliseconds between slides
     */
    public static int getDelayBetweenSlides(Context context) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
        String value = prefs.getString(PreferencesKeys.SEQUENCE_DELAY_KEY, "");
        if (TextUtils.isEmpty(value)) {
            return 0;
        } else {
            return Integer.parseInt(value);
        }
    }
}

File: Numbers-Teach-Android/app/src/main/java/com/dferreira/numbers_teach/preferences/PreferencesUtils.kt
package com.dferreira.numbers_teach.preferences

import android.content.Context
import android.preference.PreferenceManager

/**
 * Utils to provide the values of the preferences selected
 */
object PreferencesUtils {
    /**
     * @param context Get the delay betweenSlides
     * @return Delay in milliseconds between slides
     */
    fun getDelayBetweenSlides(context: Context): Int {
        val preferences = PreferenceManager.getDefaultSharedPreferences(context)
        val value = preferences.getString(PreferencesKeys.SEQUENCE_DELAY_KEY, "")
        return if (value.isNullOrEmpty()) {
            0
        } else {
            value.toInt()
        }
    }
}